tests/conftest.py

```diff
@@ -3,6 +3,7 @@
 import os
 import sys
 
+import numpy as np
```

Comment:
I know you said you're struggling to add unit tests for all these changes and I'm curious what you think is missing.  These look good to me although I'll keep thinking about it 🤔 🎉